### ISSUE TITLE
(WCAG 2.1) Add indexes to enumerator labels

### ISSUE THREAD

**Issue by jeanniefu** [civiform/civiform#2624]
Feedback from a11y experts during an a11y audit.

Severity (Critical/Serious/Moderate/Good to have): Good to have

In the below example, the placeholder text "child name" could be changed to "child # 2 name".

<img width="1270" alt="Screen Shot 2022-06-01 at 6 20 13 PM" src="https://user-images.githubusercontent.com/9094240/171527252-5a78b2d0-fcc5-4379-924a-ec85352bd704.png">

Related issue:
* https://github.com/seattle-uat/civiform/issues/2617

**Comment by jeanniefu** [comment]
In conjunction with the a11y feedback in https://github.com/seattle-uat/civiform/issues/2617 and https://github.com/seattle-uat/civiform/issues/2625, my proposal is to:
- Remove placeholder text, and add labels to the left of the input field: e.g., "child # 1 name", "child # 2 name", etc.
- Update the remove button text to "Remove child # 1", "Remove child # 2", etc.

**Comment by clouser2** [comment]
One challenge to using "1st" / "4th" language might be when translating to other languages. I'm not sure if all languages follow a similar suffix pattern as English.

**Comment by jeanniefu** [comment]
This was also flagged during an accessibility audit against WCAG 2.1. We will need to fix it to meet WCAG 2.1 guidelines.

**Comment by jeanniefu** [comment]
One corner case we'll need to handle: if someone adds some enumerators and then removes an earlier one, should we shift the index or leave as is? I think we should shift the index.

Example: Someone enters

```
1st child: Alice
2nd child: Bob
3rd child: Jim
```

Then they remove Alice. Without any changes this will become:
```
2nd child: Bob
3rd child: Jim
```

We should add some client side logic to update the text to the following:
```
1st child: Bob
2nd child: Jim
```
I don't think we have access to the translated messages from the client so we may need to pipe that through, or just go with "#number".
